src/components/forms/DateField/utils.js

```diff
@@ -1,10 +1,28 @@
-import parse from 'date-fns/parse';
+import {parseISO} from 'date-fns';
 
-export const parseDate = value => {
+const RE_PARTS = {
+  year: '(?<year>[0-9]{4})',
+  month: '(?<month>[0-2]?[0-9])',
+  day: '(?<day>[0-3]?[0-9])',
+};
```

Comment:
doesn't `\d` have the problem that it matches more than just `[0-9]` because of unicode shenanigans? At least that's the case in Python.

The idea behind this was that we can already do some eager validation during regex matching to not have to parse every time, but tbh I'm not sure how expensive that is and if it would matter a lot. So perhaps this should then be turned in:

```suggestion
const RE_PARTS = {
  year: '(?<year>[0-9]{4})',
  month: '(?<month>[0-9]{1,2})',
  day: '(?<day>[0-9]{1,2})',
};
```

Reply:
Depends on the regexp implementation. But, yes, it could possibly match more numerals than `[0-9]`.

Reply:
Checked it, it's indeed just digits `[0-9]`, so I'll simplify!